test/interpreter_functional/test_suites/run_pipeline/esaggs_timeshift.ts

```diff
@@ -223,8 +223,8 @@ export default function ({
       const expression = `
           kibana_context timeRange={timerange from='${timeRange.from}' to='${timeRange.to}'}
           | esaggs index={indexPatternLoad id='logstash-*'}
-          aggs={aggFilters id="1" filters='[{"input":{"query":"geo.src:\\"US\\" ","language":"kuery"},"label":""},{"input":{"query":"geo.src: \\"CN\\"","language":"kuery"},"label":""}]'}
-          aggs={aggFilters id="2" filters='[{"input":{"query":"geo.dest:\\"US\\" ","language":"kuery"},"label":""},{"input":{"query":"geo.dest: \\"CN\\"","language":"kuery"},"label":""}]'}
+          aggs={aggFilters id="1" filters={queryFilter {kql "geo.src:\\"US\\" "}} filters={queryFilter {kql "geo.src: \\"CN\\""}}}
```

Comment:
Following up on my comment earlier about the confusing naming, it's very weird to me that you have `aggFilters`, `filters`, `queryFilter` and `kql` all combined here. Can you simplify this in any way?

Reply:
But that is semantically and logically correct. We have filters-based aggregation, and that requires filters on input. For now, we support query-based filters, so we pass a query filter as an argument. And the query filter can either be `kql` or `lucene`. That seems pretty straightforward.
Do you have any suggestions or ideas on how we can improve that? I can only think about replacing `filters` with a singular `filter`, but I kept that because the argument was already there.

Reply:
@wylieconlon with this PR we hope to remove passing json values as arguments to expression functions, so previous json filter structure is replaced by a function that can generate that same structure.

we took the same approach with aggconfigs (not long back esaggs accepted a single aggs json object, now we have functions that can generate any of the aggs). And yes, techically this is a breaking change but as we are not storing expressions using this anywhere but are rather generating them on-the-fly this should be ok.

Reply:
My main comment here and above is that `filter` and `query` are synonyms in the Kibana UI. So yes, I have a suggestion for improving the readability of this. You could either:

a. Have two parallel arrays containing `filters` & `labels`, where each label is assigned to the corresponding filter. If you do this, you could also remove the `queryFilter` function.
b. Allow each `kql` or `lucene` argument to be labeled, instead of creating a new `queryFilter` wrapper.

Reply:
we'll go forward with current approach as we actually have a different structure in kibana representing a queryFilter. Please open an issue if you think this should be refactored (as we'll need to change those types as well) and we can do that as a follow up.